part_2.md

```diff
@@ -143,22 +133,37 @@ You'll see in the next part how we can add a valid authentication token to the r
+To generate a valid token we first need to create a second app registration in the Azure portal. This is to register the application which will be requesting access to the API. 
+
+Create a new app registration called `WeatherAppConsumer` (feel free to name it something else if you prefer!).
+* Once created, add a new **client secret** from the **Certificates & secrets**
+
+Next you'll need to grant API permission for the new application to access the first app registration you created.
+* Select **API permissions** => **Add a permission** => **My APIs** then click on your app and the role you created earlier (`WeatherApplicationRole`) 
+* **Make sure you grant admin consent for the tenant on the API permissions page**. As you created the tenant you should have admin permissions to do so.
 
 ### 2.2: Get a token to access the web API
 
-You should now be able to request a token to access the API. You can do this by just using `curl` in the terminal. See [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow#first-case-access-token-request-with-a-shared-secret) for what the request should look like. In particular:
+You should now be able to request a token to access the API. See [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow#first-case-access-token-request-with-a-shared-secret) for what the request should look like. 
+
+The structure of the request in Postman will look like the following:
+
+![Token Request](WeatherForecast/img/TokenRequest.PNG)
+
+In particular:
 
 - The tenant id should be from the tenant you created in part 1. You can find this on the overview page for either of the app registrations you've created.
 - The client id should be the client id for the app registration you created in step 2.1.
-- The scope should be the application ID URI from the first app registration you created in step 1.3 followed by "/.default" and it needs to be URI encoded. For example `api%3A%2F%2F40ae91b7-0c83-4b5c-90f3-40187e8f2cb6%2F.default` would be the correct scope for application ID URI api://40ae91b7-0c83-4b5c-90f3-40187e8f2cb6. You can find the application ID URI by going to the "Expose an API" section for your first app registration in the Azure portal.
+- The scope should be the application ID URI from the first app registration you created in step 1.3 followed by "/.default" and it needs to be URI encoded. For example `api%3A%2F%2F40ae91b7-0c83-4b5c-90f3-40187e8f2cb6%2F/.default` would be the correct scope for application ID URI api://40ae91b7-0c83-4b5c-90f3-40187e8f2cb6. You can find the application ID URI by going to the "Expose an API" section for your first app registration in the Azure portal.

```

Comment:
The /  in this line is encoded as %2F, so I think this one's an error
```suggestion
- The scope should be the application ID URI from the first app registration you created in step 1.3 followed by "/.default" and it needs to be URI encoded. For example `api%3A%2F%2F40ae91b7-0c83-4b5c-90f3-40187e8f2cb6%2F.default` would be the correct scope for application ID URI api://40ae91b7-0c83-4b5c-90f3-40187e8f2cb6. You can find the application ID URI by going to the "Expose an API" section for your first app registration in the Azure portal.
```

Reply:
Ah good spot this is what I get for using postman (probably need to add a comment around that).